.circleci/config.yml

```diff
@@ -229,7 +229,7 @@ jobs:
           command: |
             python -m venv venv
             . venv/bin/activate
-            git clone --depth 1 https://github.com/plotly/dash.git -b dev dash-main
+            git clone --depth 1 https://github.com/plotly/dash.git -b ${CIRCLE_BRANCH} dash-main
```

Comment:
oh hmm - I hadn't really quite paid attention to what's going on here before. This will work (clearly, since it works 😏) so if you're itching to merge and move on that's fine. But if you do leave it as is, please make an issue to come back to this:
- at the very least we already have this code via `checkout` so could use that instead of cloning again?
- but I wonder if we can't piggyback on the `build-core-37` job that already does most of the cloning and building you're doing here, the same way that `test-37` does.
- `build-core-37` also has a line `sed -i '/dash-/d' requires-install.txt` that prevents bringing in a second copy of dcc, html, and table - if we *don't* use the results of `build-core-37` we may want to do the same here.

Reply:
@alexcjohnson I wasn't able to reuse the `dash` directory from an earlier run, though I can't recall why -- I did try that.

FWIW these lines are essentially pulled straight from https://github.com/plotly/dashR/blob/9c5a30760e3be985c866f122e9c2bd6ce2a516a6/.circleci/config.yml#L41-L44, though it's a bit different there since we aren't running *all* the Dash tests in that configuration, and we do clone the repo earlier in this workflow.

Reply:
Sure, that's fine - just make an issue to come back to this.

Reply:
Since these are separate jobs, the paths are not shared by default, as far as I am aware. We could `persist_to_workspace` the `dash` package directory from a preceding run, so I'll open an issue to suggest this. We already do this for package tarballs, though not the repos themselves.

https://circleci.com/docs/2.0/workflows/#using-workspaces-to-share-artifacts-among-jobs

I am not certain that we want to reuse the component library directories after deleting most of their content for the R test run, but maybe this is a non-issue if we don't subsequently `persist_to_workspace` those paths. I'm not sure, I'm pretty new to configuring CircleCI.

Reply:
@alexcjohnson https://github.com/plotly/dash/issues/1264